pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -722,6 +722,9 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 		}
 
 		svcConv := kube.ConvertService(*svc, c.domainSuffix, c.ClusterID)
+		if len(svcConv.Ports) == 0 {
```

Comment:
Are you sure we don't need to do anything below, even if we don't have ports? Why not just always do everything, even for headless?